js/ai/src/generate.ts

```diff
@@ -372,10 +374,12 @@ export function generateStream<
 ): GenerateStreamResponse<O> {
   let channel = new Channel<GenerateResponseChunk>();
 
-  const generated = generate<O, CustomOptions>(registry, {
-    ...options,
-    onChunk: (chunk) => channel.send(chunk),
-  });
+  const generated = Promise.resolve(options).then((resolvedOptions) =>
+    generate<O, CustomOptions>(registry, {
+      ...resolvedOptions,
+      onChunk: (chunk) => channel.send(chunk),
+    })
+  );
```

Comment:
What's this about? Probably fine just curious.

Reply:
This is actually a sort of unrelated bug fix -- when generateStream was made sync we broke the ability to accept options as a promise. This fixes that.